src/v1/device_manager_client.ts

```diff
@@ -1388,27 +1394,69 @@ export class DeviceManagerClient {
       {} | undefined
     ]
   > | void {
-    request = request || {};
-    let options: CallOptions;
-    if (typeof optionsOrCallback === 'function' && callback === undefined) {
-      callback = optionsOrCallback;
-      options = {};
-    } else {
-      options = optionsOrCallback as CallOptions;
-    }
-    options = options || {};
-    options.otherArgs = options.otherArgs || {};
-    options.otherArgs.headers = options.otherArgs.headers || {};
-    options.otherArgs.headers['x-goog-request-params'] =
-      this._gaxModule.routingHeader.fromParams({
-        name: request.name || '',
+
+    return new Promise(async (resolve, reject) => {
+      const token_response = await this.getRegistryToken();
+      const token = JSON.parse(token_response);
+      const payload = JSON.stringify({
+        name: request?.name,
+        numVersions: request?.numVersions
+      })
+      var options = {
+        host: 'iot-sandbox.clearblade.com',
+        path: `/api/v/1/code/` + token.systemKey + `/devicesConfigVersionsList`,
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'ClearBlade-UserToken': token.serviceAccountToken,
+          'Content-Length': payload.length
+        }
+      };
+
+      const req = https.request({
+        ...options,
+      }, res => {
+        let data = '';
+        const chunks: any[] = [];
+        res.on('data', chunk => data += chunk)
+        res.on('end', () => {
+          if (!this.isJsonString(data)) {
+            reject(data);
+            return;
+          }
+          let array: [protos.google.cloud.iot.v1.IListDeviceConfigVersionsResponse, protos.google.cloud.iot.v1.IListDeviceConfigVersionsRequest | undefined, {} | undefined];
+
+          const ilistdeviceconfigversionsresponse: protos.google.cloud.iot.v1.IListDeviceConfigVersionsResponse = {};
+
+          const ideviceconfigarray: protos.google.cloud.iot.v1.IDeviceConfig[] | null = [];
+
+          const configList = JSON.parse(data);
+          //for loop fetching JSON
+          for (let index in configList.deviceConfigs) {
```

Comment:
nitpick: this code could be simplied by using Array.map

Reply:
Sure, will improve that in next JIRA.